src/shadowbox/server/mocks/mocks.ts

```diff
@@ -78,3 +35,35 @@ export class InMemoryFile implements TextFile {
     this.exists = true;
   }
 }
+
+export class MockShadowsocksServer implements ShadowsocksServer {
+  private accessKeys: ShadowsocksAccessKey[] = [];
+
+  update(keys: ShadowsocksAccessKey[]) {
+    this.accessKeys = keys;
+    return Promise.resolve();
+  }
+
+  getAccessKeys() {
+    return this.accessKeys;
+  }
+}
+
+export class ManagerMetricsStub implements ManagerMetrics {
```

Comment:
I believe this is also a fake by Google terminology, since for that a stub "only returns what you tell it to return", whereas this is a shallow implementation of the interface.

Reply:
Renamed in a9bd8a5.